src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -22,159 +28,107 @@
-                    final BufferedReader errorBr = new BufferedReader(new InputStreamReader(error));
-                    sb.append("\"Error Body\": \"");
-                    String line = errorBr.readLine();
-                    while (line != null) {
-                        sb.append(line).append(System.lineSeparator());
-                        line = errorBr.readLine();
-                    }
-                    errorBr.close();
-                } else {
-                    sb.append("Unable to open error stream for reading");
-                }
-                sb.append("\"");
+                sb.append("\"Code\": ").append(response.code()).append(",");
+                sb.append("\"Message\": \"").append(response.message()).append("\",");
+                sb.append("\"Error Body\": \"").append(responseBody.string()).append("\"");
+                sb.append('}');
+                sb.append('}');
+                throw new IOException(String.valueOf(response));
             }
-
-            sb.append("}");
-            sb.append("}");
-            log.error("Encountered connection error: " + sb);
+            return objectMapper.readTree(responseBody.string());
+        } catch (IOException e) {
             throw e;
-        } finally {
-            if (br != null) {
-                try {
-                    br.close();
-                } catch (final IOException ignored) {
-                }
-            }
-        }
-    }
-
-    private HttpsURLConnection getURLConnection(final String urlString) throws IOException {
-        final URL url = new URL(urlString);
-        final HttpsURLConnection urlConnection = (HttpsURLConnection) url.openConnection();
-        urlConnection.setRequestProperty("Authorization", authentication);
-
-        if(cookies.length() > 0) {
-            urlConnection.setRequestProperty("Cookie", cookies);
         }
-        return urlConnection;
-    }
-
-    private void setCookies() {
-        final StringBuilder sb = new StringBuilder();
-        if(jsessionId != null) {
-            sb.append("JSESSIONID=").append(jsessionId);
-        }
-        if(upstream != null) {
-            if(sb.length() > 0) {
-                sb.append(";");
-            }
-            sb.append("upstream=").append(upstream);
-        }
-        cookies = sb.toString();
     }
 
     private String getBasicAuth() {
         final String userPass = config.getJiraUsername() + ":" + config.getJiraPassword();
         final String basicAuth = "Basic " + new String(new Base64().encode(userPass.getBytes()));
         return basicAuth;
     }
+
+    private static CookieJar setupCookieJar() {
+        return new CookieJar() {
+            private final Map<String, Cookie> cookies = new HashMap<>();
+            @Override
+            public void saveFromResponse(final HttpUrl url, final List<Cookie> cookies) {
+                cookies.forEach(cookie -> {
+                    this.cookies.remove(cookie.name());
```

Comment:
I know I suggested this, but looking at it again, I think the remove() call is redundant.